run_galaxy_workflow.py

```diff
@@ -193,8 +193,12 @@ def main():
             results = state.results
 
         # Produce tool versions file
+        tool_table_path = "{}/software_versions_galaxy.txt".format(args.output_dir)
+        # remove file if exists already
+        if path.exists(tool_table_path):
+            remove(tool_table_path)
```

Comment:
I would rather have this whole logic living inside the function, so that the overall flow is cleaner. But I can live with this if it is a complication to chuck it inside (I could think that you could decide to do it if tool_list being passed is empty, although it is a bit nasty (but it signals the initial phase of the recursion, where you want to be doing this).